src/Dfc.CourseDirectory.WebV2/Features/Courses/DeleteCourseRun.cs

```diff
@@ -0,0 +1,168 @@
+            ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        {
+            _providerInfoCache = providerInfoCache;
+            _providerOwnershipCache = providerOwnershipCache;
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+        }
+
+        public async Task<ViewModel> Handle(Request request, CancellationToken cancellationToken)
+        {
+            var ukprn = await GetUkprnForCourse(request.CourseId);
+            var (course, courseRun) = await GetCourseAndCourseRun(request.CourseId, request.CourseRunId, ukprn);
+            return await CreateViewModel(course, courseRun);
+        }
+
+        public async Task<OneOf<ModelWithErrors<ViewModel>, SuccessViewModel>> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var ukprn = await GetUkprnForCourse(request.CourseId);
+            var (course, courseRun) = await GetCourseAndCourseRun(request.CourseId, request.CourseRunId, ukprn);
+
+            // User must have confirmed
+            if (!request.Confirm)
+            {
+                var vm = await CreateViewModel(course, courseRun);
+                var validationResult = new ValidationResult(new[]
+                {
+                    new ValidationFailure(nameof(request.Confirm), "Confirm you want to delete the course")
+                });
+                return new ModelWithErrors<ViewModel>(vm, validationResult);
+            }
+
+            await _cosmosDbQueryDispatcher.ExecuteQuery(new DeleteCourseRunQuery()
+            {
+                CourseId = request.CourseId,
+                CourseRunId = request.CourseRunId,
+                ProviderUkprn = ukprn
+            });
+
+            return new SuccessViewModel()
+            {
+                CourseId = request.CourseId,
+                CourseRunId = request.CourseRunId,
+                CourseName = courseRun.CourseName
+            };
+        }
+
+        private async Task<ViewModel> CreateViewModel(Course course, CourseRun courseRun)
+        {
+            Venue venue = default;
+            if (courseRun.VenueId.HasValue)
+            {
+                venue = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetVenueById() { VenueId = courseRun.VenueId.Value });
+            }
+
+            return new ViewModel()
+            {
+                CourseId = course.Id,
+                CourseRunId = courseRun.Id,
+                CourseName = courseRun.CourseName,
+                DeliveryMode = courseRun.DeliveryMode,
+                FlexibleStartDate = courseRun.FlexibleStartDate,
+                StartDate = courseRun.StartDate,
+                VenueName = venue?.VenueName,
+                YourReference = courseRun.ProviderCourseID
+            };
+        }
+
+        private async Task<(Course Course, CourseRun CourseRun)> GetCourseAndCourseRun(Guid courseId, Guid courseRunId, int ukprn)
+        {
+            (await _cosmosDbQueryDispatcher.ExecuteQuery(new GetCoursesByIds()
```

Comment:
```suggestion
            (await _cosmosDbQueryDispatcher.ExecuteQuery(new GetCoursesByIds
```
remove redundant parenthethethis on object instantiation